testng-core/src/main/java/org/testng/JarFileUtils.java

```diff
@@ -76,6 +76,10 @@ private boolean testngXmlExistsInJar(File jarFile, List<String> classes) throws
         if (Parser.canParse(jeName.toLowerCase())) {
           InputStream inputStream = jf.getInputStream(je);
           File copyFile = new File(file, jeName);
+          File parent = copyFile.getParentFile();
+          if (!parent.exists()) {
```

Comment:
Hi, I think my solution is better. mkdirs already check if directory exists so there is no need to check it. 
file.mkdirs()

Reply:
Fixed